checks/branch_protection.go

```diff
@@ -153,115 +165,122 @@ func isBranchProtected(branches []*github.Branch, name string) (bool, error) {
-func requiresStatusChecks(protection *github.Protection, branch string, dl checker.DetailLogger) bool {
-	// This is disabled by default (bad).
-	if protection.GetRequiredStatusChecks() != nil &&
-		protection.RequiredStatusChecks.Strict &&
-		len(protection.RequiredStatusChecks.Contexts) > 0 {
-		return true
+// Maximum score returned is 2.
+func requiresStatusChecks(protection *github.Protection, branch string, dl checker.DetailLogger) int {
+	score := 0
+
+	if protection.GetRequiredStatusChecks() == nil ||
+		!protection.RequiredStatusChecks.Strict {
+		dl.Warn("status checks for merging disabled on branch '%s'", branch)
+		return score
 	}
-	switch {
-	case protection.RequiredStatusChecks == nil ||
-		!protection.RequiredStatusChecks.Strict:
-		dl.Warn("Status checks for merging disabled on branch '%s'", branch)
-	case len(protection.RequiredStatusChecks.Contexts) == 0:
-		dl.Warn("Status checks for merging have no specific status to check on branch '%s'", branch)
-	default:
-		panic("Unhandled status checks error")
+
+	dl.Info("strict status check enabled on branch '%s'", branch)
+	score++
+
+	if len(protection.RequiredStatusChecks.Contexts) > 0 {
+		dl.Warn("status checks for merging have specific status to check on branch '%s'", branch)
+		score++
+	} else {
+		dl.Warn("status checks for merging have no specific status to check on branch '%s'", branch)
 	}
-	return false
+
+	return score
 }
 
 // Returns true if several PR review requirements are enabled. Otherwise returns false and logs why it failed.
-func requiresThoroughReviews(protection *github.Protection, branch string, dl checker.DetailLogger) bool {
-	// This is disabled by default (bad).
-	if protection.GetRequiredPullRequestReviews() != nil &&
-		protection.RequiredPullRequestReviews.RequiredApprovingReviewCount >= minReviews &&
-		protection.RequiredPullRequestReviews.DismissStaleReviews &&
-		protection.RequiredPullRequestReviews.RequireCodeOwnerReviews {
-		return true
-	}
-	if protection.RequiredPullRequestReviews == nil {
-		dl.Warn("Pullrequest reviews disabled on branch '%s'", branch)
-		return false
+// Maximum score returned is 5.
+func requiresThoroughReviews(protection *github.Protection, branch string, dl checker.DetailLogger) int {
+	score := 0
+
+	if protection.GetRequiredPullRequestReviews() == nil {
+		dl.Warn("pull request reviews disabled on branch '%s'", branch)
+		return score
 	}
-	switch {
-	case protection.RequiredPullRequestReviews.RequiredApprovingReviewCount < minReviews:
-		dl.Warn("Number of required reviewers is only %d on branch '%s'",
+
+	if protection.RequiredPullRequestReviews.RequiredApprovingReviewCount >= minReviews {
+		dl.Info("number of required reviewers is %d on branch '%s'",
 			protection.RequiredPullRequestReviews.RequiredApprovingReviewCount, branch)
-		fallthrough
-	case !protection.RequiredPullRequestReviews.DismissStaleReviews:
+		score += 2
```

Comment:
2 -> minReviews?

Reply:
2 reviews means it makes compromised contributors/sockpuppet attacks harder. 
Do we want to encourage devs to have more than 2 reviews? If not, then I think 2 is enough. Let me know what you think.

Reply:
oh nope! i just figured you may want to use the variable minReviews = 2
oops, i think the coder notation is like
s/2/minReviews